### ISSUE TITLE
INSERT .. RETURNING returns wrong binary data

### ISSUE THREAD

**Issue by lukaseder** [FirebirdSQL/jaybird#728]
Here's a self contained test case that currently fails with jaybird 4.0.8:

```java
try (Statement s = connection.createStatement()) {
    try {
        s.executeUpdate("create table t (a integer not null primary key, b blob)");

        try (ResultSet rs = s.executeQuery("insert into t (a, b) values (1, X'010203') returning a, b")) {
            while (rs.next())
                System.out.println(Arrays.toString(rs.getBytes(2)) + ": " + rs.getBytes(2).length);
        }

        try (ResultSet rs = s.executeQuery("select a, b from t")) {
            while (rs.next())
                System.out.println(Arrays.toString(rs.getBytes(2)) + ": " + rs.getBytes(2).length);
        }
    }
    finally {
        s.executeUpdate("drop table t");
    }
}
```

It prints:

```
[0, 0, 40, -60, 0, 0, 0, 0]: 8
[1, 2, 3]: 3
```

The first result isn't even stable, e.g. after executing it again, I'm getting:

```
[0, 0, 40, -58, 0, 0, 0, 0]: 8
[1, 2, 3]: 3
```

And then:

```
[0, 0, 40, -57, 0, 0, 0, 0]: 8
[1, 2, 3]: 3
```

**Comment by mrotteveel** [comment]
Thanks for reporting. The value returned seems to be the blob-id instead of the blob value, which would suggest that this is backed by an `FBCachedBlob` which was populated with the blob-id and not the blob value. I'll see if I can trace where this occurs.

**Comment by mrotteveel** [comment]
The code path taken intentionally does not fetch the blob value, but I'm unsure why.

**Comment by mrotteveel** [comment]
Fix will be available with 4.0.9 and 5.0.1. I don't have a solid release date for either of those yet, but probably somewhere around the beginning of February, maybe earlier if it turns out I broke a lot in 5.0.0 ;).

You can test with the 5.0.1.java11-SNAPSHOT (or 5.0.1.java8-SNAPSHOT) on https://oss.sonatype.org/content/repositories/snapshots (deploying 4.0.x snapshots is a lot more manual work, so I generally won't bother, but let me know if you prefer to have a 4.0.9 snapshot).

**Comment by lukaseder** [comment]
I can wait for 4.0.9, thanks!
